### DISCUSSION TITLE
Is it possible to extract order by clauses / filter clauses / selected columns from a SelectStatement?

### DISCUSSION THREAD

**Discussion by vamshiaruru-virgodesigns** [go-jet/jet#221] [Q&A]
Hi everyone, like the title says, given a SelectStatement? The usecase is that I want to implement keyset/cursor based pagination. Ideally, I want to append the new order by clauses and where clauses that the keyset pagination requires, but if a query already has an order by clause / where clause set, adding a new order by  / whereseems to overwrite the previous order by / where by. So I was thinking if I can find what order by/ where clauses the query has, I can create a new order by/where clause by appending the keyset clauses to the existing ones. 

Also, are you guys aware of any keyset pagination implementations that use jet? 

Thanks in advance!

**Comment by go-jet** [comment]
Hi @vamshiaruru-virgodesigns ,
It is not possible to extract clauses from the statement. Instead you can leave creation of statement as a last step, and modify clauses before they are passed to statement.
Here is a pagination sample: https://github.com/go-jet/jet/discussions/148#discussioncomment-2745062
